src/scan_nut.cc

```diff
@@ -250,29 +380,10 @@ scan_nut_actor(zsock_t *pipe, void *args)
         std::vector<std::string> output;
         nut_scan_multi_xml_http ("device", listAddr->firstAddress(), listAddr->lastAddress(), output);
         if (! output.empty ()) {
-            s_nut_output_to_fty_messages (&listDiscovered , output, devices);
-
-            for (auto asset:listDiscovered) {
-                if(ask_actor_term(pipe)) stop_now = true;
-                if(zsys_interrupted || stop_now) {
-                    fty_proto_destroy(&asset);
-                } else {
-                    fty_proto_t *msg = asset;
-                    std::string addr = fty_proto_ext_string(msg, "ip.1", "");
-
-                    map_string_t nutdata;
-                    if (nut_dumpdata_netxml_ups ("http://"+addr, nutdata) == 0) {
-                        s_nut_dumpdata_to_fty_message (msg, nutdata);
-                    }
-
-                    zmsg_t *reply = fty_proto_encode (&msg);
-                    zmsg_pushstr(reply, "FOUND");
-                    zmsg_send (&reply, pipe);
-                }
-            }
-            listDiscovered.clear();
+            stop_now = create_pool_dumpdata(output, devices, pipe, "", "xml");
         }
```

Comment:
this code looks missing compare to snmp loop, doesn't it ?
 if(ask_actor_term(pipe)) stop_now = true;
            if(zsys_interrupted || stop_now)
                break;